includes/base/controls-stack.php

```diff
@@ -393,6 +393,12 @@ public function add_control( $id, array $args, $options = [] ) {
 			'position' => null,
 		];
 
+		$should_optimize_controls = $this->should_optimize_controls();
+
+		if ( $should_optimize_controls && ! empty( $args['type'] ) && in_array( $args['type'], [ Controls_Manager::RAW_HTML, Controls_Manager::DIVIDER, Controls_Manager::HEADING, Controls_Manager::BUTTON ] ) ) {
+			return false;
```

Comment:
Can we not add this code totally at the beginning of this method? `$default_options` is redundant if we return early here.